gateway/ln-gateway/src/gateway.rs

```diff
@@ -44,29 +50,43 @@ impl Gateway {
     pub async fn new(
         config: GatewayConfig,
         decoders: ModuleDecoderRegistry,
-        ln_rpc: Arc<dyn LnRpc>,
+        lnrpc_factory: DynLnRpcClientFactory,
```

Comment:
This argument could just be a `DynLnRpcCLient`. We could still inject a mock client that way.